tutorials/physics/using_jolt_physics.rst

```diff
@@ -0,0 +1,473 @@
+Setting this project setting to ``1.0`` will effectively disable the technique,
+resulting in a simulation that more closely resembles Godot Physics, but which is
+also more unstable.
+
+Motion queries (move_and_slide, etc.)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Physics servers in Godot are meant to implement the :ref:`PhysicsServer3D.body_test_motion()<class_PhysicsServer3D_method_body_test_motion>`
+method, which in turn powers methods like :ref:`PhysicsBody3D.test_move()<class_PhysicsBody3D_method_test_move>`,
+:ref:`PhysicsBody3D.move_and_collide()<class_PhysicsBody3D_method_move_and_collide>`
+and :ref:`CharacterBody3D.move_and_slide()<class_CharacterBody3D_method_move_and_slide>`,
+which are largely meant to be used for moving player characters around.
+
+:ref:`PhysicsServer3D.body_test_motion<class_PhysicsServer3D_method_body_test_motion>`
+is split into three parts, the first being depenetration (called "recovery" in
+Godot), the second being a shape-cast from the "recovered" position, and the third
+being the actual collision check, at the "unsafe" fraction of the shape-cast.
+
+The "recovery" step in Godot Physics is hardcoded to always do 4 iterations with 40%
+depenetration per iteration. I figured these constants might be useful to expose, so
+when using this module they can be configured in the project settings as
+:ref:`physics/jolt_physics_3d/motion_queries/recovery_iterations<class_ProjectSettings_Property_physics/jolt_physics_3d/motion_queries/recovery_iterations>`
+and :ref:`physics/jolt_physics_3d/motion_queries/recovery_amount<class_ProjectSettings_Property_physics/jolt_physics_3d/motion_queries/recovery_amount>`
+respectively.
+
+The implementation of :ref:`PhysicsServer3D.body_test_motion()<class_PhysicsServer3D_method_body_test_motion>`
+in this module also differs slightly from the Godot Physics implementation, as
+replicating the Godot Physics version resulted in a prohibitive amount of ghost
+collisions for :ref:`move_and_slide<class_CharacterBody3D_method_move_and_slide>`.
+
+The discrepancies are as follows:
+
+- There is no TEST_MOTION_MIN_CONTACT_DEPTH_FACTOR.
+- There is no epsilon added to the safe margin during recovery.
+- Contacts with normals that are not opposing the motion vector are discarded from
+  the final collision check.
+
+While these discrepancies do seem to result in less ghost collisions when using
+Jolt, they also introduce new problems:
+
+- :ref:`move_and_slide<class_CharacterBody3D_method_move_and_slide>` is generally
+  slower than in Godot Physics, due to :ref:`class_CharacterBody3D`::_snap_on_floor(PR NOTE: Apply_floor_snap?)
+  triggering on every call.
+- :ref:`class_CharacterBody3D` will not report wall collisions unless you're moving towards them.
+
+It's not clear to me how to fix these issues, but I suspect they will require
```

Comment:
Do we want to use 'I' in documentation?

Reply:
Same as the previous comment. This entire section should be removed in my opinion.